hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ChangedReadersObserver.java

```diff
@@ -40,5 +40,16 @@
    * @param memStoreScanners scanner of current memstore
    * @throws IOException e
    */
-  void updateReaders(List<HStoreFile> sfs, List<KeyValueScanner> memStoreScanners) throws IOException;
+  void updateReaders(List<HStoreFile> sfs, List<KeyValueScanner> memStoreScanners)
```

Comment:
Create a POJO and use it to pass the list of store files, list of memstoreScanners and a boolean to indicate if it is flush/compaction. Act accordingly in the Storesscanner? It will avoid having duplicate methods. Down in StoreScanner the volatile can be renamed accordingly?

Reply:
Created POJO for the method signature and same is to be used by flush and compaction.
However, since there are some differences in locking (e.g. using closeLock for flush but not for compaction), it might be good to have private methods for flush and compaction(to be used by main method).
Updated the PR @ramkrish86